rattan-core/src/radix/mod.rs

```diff
@@ -33,14 +32,30 @@ use crate::{control::http::HttpControlEndpoint, error::HttpServerError};
 #[cfg(feature = "http")]
 use std::net::{Ipv4Addr, SocketAddr};
 
-pub mod log;
-
-pub use log::RattanLogOp;
-
 pub static INSTANCE_ID: OnceCell<String> = OnceCell::new();
-pub static LOGGING_TX: OnceCell<UnboundedSender<RattanLogOp>> = OnceCell::new();
 pub static BASE_TS: OnceCell<i64> = OnceCell::new();
 
+use rattan_log::{file_logging_thread, RattanLogOp, LOGGING_TX};
+#[cfg(feature = "serde")]
+#[derive(
+    Clone, Copy, Debug, clap::ValueEnum, PartialEq, Eq, serde::Deserialize, serde::Serialize,
+)]
+pub enum PacketLogMode {
+    CompactTCP,
+    RawIP,
+    RawTCP,
+}
+
+#[cfg(not(feature = "serde"))]
+#[derive(Clone, Copy, Debug, clap::ValueEnum, PartialEq, Eq)]
+pub enum PacketLogMode {
+    CompactTCP,
+    RawIP,
+    RawTCP,
+}
```

Comment:
Do not define the struct twice. You should use `#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]` instead.